alchemy/bin/commands/deploy.ts

```diff
@@ -0,0 +1,216 @@
+      ),
+    );
+    log.info("Create an alchemy.run.ts file to define your infrastructure.");
+    process.exit(1);
+  }
+
+  // Detect package manager
+  const packageManager = detectPackageManager(path);
+
+  // Build command arguments
+  const args: string[] = [];
+  if (quiet) args.push("--quiet");
+  if (read) args.push("--read");
+  if (destroy) args.push("--destroy");
+  if (stage) args.push(`--stage ${stage}`);
+
+  const argsString = args.join(" ");
+
+  // Determine the command to run based on package manager and file extension
+  let command: string;
+  const isTypeScript = runFile.endsWith(".ts");
+
+  switch (packageManager) {
+    case "bun":
+      command = `bun ${runFile} ${argsString}`;
+      break;
+    case "deno":
+      command = `deno run -A ${runFile} ${argsString}`;
+      break;
+    case "pnpm":
+      command = isTypeScript
+        ? `pnpm tsx ${runFile} ${argsString}`
+        : `pnpm node ${runFile} ${argsString}`;
+      break;
+    case "yarn":
+      command = isTypeScript
+        ? `yarn tsx ${runFile} ${argsString}`
+        : `yarn node ${runFile} ${argsString}`;
+      break;
+    default:
+      command = isTypeScript
+        ? `npx tsx ${runFile} ${argsString}`
+        : `node ${runFile} ${argsString}`;
+      break;
+  }
+
+  // Execute the deploy
+  async function executeDeploy() {
+    const s = spinner();
+    s.start(`Executing: ${command}`);
+
+    try {
+      const { stdout, stderr } = await execAsync(command, {
+        cwd: path,
+        env: {
+          ...process.env,
+          FORCE_COLOR: "1",
+        },
+      });
+
+      s.stop("Deploy command executed");
+
+      if (stdout) {
+        console.log(stdout);
+      }
+
+      if (stderr) {
+        console.error(stderr);
+      }
+    } catch (error: any) {
+      s.stop("Deploy failed");
+      log.error(pc.red(`Deploy failed: ${error.message}`));
+      if (error.stdout) {
+        console.log(error.stdout);
+      }
+      if (error.stderr) {
+        console.error(error.stderr);
+      }
+      if (!watchMode) {
+        process.exit(1);
+      }
+    }
+  }
+
+  // Initial deploy
+  await executeDeploy();
+
+  // Set up watch mode if requested
+  if (watchMode) {
+    const watchDir = dirname(runFile);
+    const watchers: any[] = [];
+    let debounceTimer: NodeJS.Timeout | null = null;
+
+    // Function to handle file changes
+    const handleFileChange = (filename?: string) => {
+      // Debounce to prevent multiple rapid redeployments
+      if (debounceTimer) {
+        clearTimeout(debounceTimer);
+      }
+
+      debounceTimer = setTimeout(async () => {
+        if (filename) {
+          log.info(pc.yellow(`\nFile changed: ${filename}`));
+        }
+        log.info(pc.yellow("Redeploying..."));
+        await executeDeploy();
+        log.info(pc.cyan("\nWatching for changes..."));
+      }, 300);
+    };
+
+    // Watch the main alchemy.run file
+    const mainWatcher = watch(runFile, (eventType) => {
```

Comment:
For watch we should be passing --watch instead of watching the file. 

Their script can be spread across many files so we want to rely on the runtime's watch which knows which modules have been loaded and need to be watched.

Reply:
I think we can always drop the `--watch` portion of this for now.